test/threadsafe_priority_queue_unittest.cc

```diff
@@ -0,0 +1,74 @@
+
+// Copyright 2017 Husky Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <map>
+#include <string>
+#include <utility>
+#include <cstdlib>
+#include <vector>
+
+#include "gtest/gtest.h"
+
+#include "husky-base/threadsafe_priority_queue.h"
+#include "husky-base/mailbox_types.h"
+
+// TODO This contains unittests for different components of Mailbox. They should be separated later.
+
+namespace husky {
+namespace base {
+namespace {
+
+class TestEventQueue : public testing::Test {
+ public:
+  TestEventQueue() {}
+  ~TestEventQueue() {}
+
+ protected:
+  void SetUp() {}
+  void TearDown() {}
+};
+
+TEST_F(TestEventQueue, InitDestroy) {
+  ThreadsafePriorityQueue<std::shared_ptr<MailboxEventBase>> queue;
+}
+
+TEST_F(TestEventQueue, Functional) {
```

Comment:
Looks like it's lack of multiple threads case.

Reply:
Yes...But it seems that it is hard to verify the result in multiple threads case if we treat it like a black box. Do you have any suggestion? Or test it in h4?

Reply:
Test case is relative to test coverage. If this is hard to come up with now, we can implement in the future. Not an urgent issue.